lib/kafka/protocol/message.rb

```diff
@@ -81,7 +104,18 @@ def self.decode(decoder)
         new(key: key, value: value, codec_id: codec_id, offset: offset)
       end
 
-      private
+      def self.v2_decode(offset, message_decoder)
+        attributes = message_decoder.int8
+        timestamp = message_decoder.int64
```

Comment:
The timestamp should probably be 1) converted from ms to s, and 2) wrapped in a Time object.